### PR TITLE
Move test/jit/test_onnx_export.py to test/onnx

### LINKED ISSUE #75627: Move test/jit/test_onnx_export.py to test/onnx
This file should be moved to the test/onnx dir:
test/jit/test_onnx_export.py

Perhaps combined with test/onnx/test_pytorch_onnx_no_runtime.py which is being added in https://github.com/pytorch/pytorch/pull/73284.

That will let us merge changes to that file without going through the PyTorch internal tests, and it will also just consolidate all the ONNX tests for easier discoverability.

### PR DESCRIPTION
**Pull request by titaiwangms**
Fixes #75627
merged test/jit/test_onnx_export.py into test/onnx/test_pytorch_onnx_no_runtime.py

### CODE REVIEW

File: test/onnx/test_pytorch_onnx_no_runtime.py [hunk 1]
```diff
@@ -4,20 +4,22 @@
...
 import unittest
-from typing import Optional, Type
+from typing import Dict, Optional, Type
 
 import onnx
 
 import torch
 from torch import Tensor
-from torch.onnx import symbolic_helper
+from torch.onnx import OperatorExportTypes, symbolic_helper, utils
```
Comment (justinchuby) [inline test/onnx/test_pytorch_onnx_no_runtime.py:13]: I would avoid importing a class or a function from a module. (https://github.com/pytorch/pytorch/wiki/torch.onnx-Namespacing)

Import only the module instead to keep the name space clean and provide readers with more context on where its members come from. For example, readers don’t need to backtrack to see `sleep` is a function from `time`, as opposed to a function defined in the file. https://google.github.io/styleguide/pyguide.html#22-imports

Comment (justinchuby) [reply test/onnx/test_pytorch_onnx_no_runtime.py:13]: In this case, `import torch._C._onnx as _C_onnx` and use `_C_onnx.OperatorExportTypes`

**Review by justinchuby — COMMENTED** [review]
Run `lintrunner -a` to fix formatting.

File: test/onnx/test_pytorch_onnx_no_runtime.py (hunk 1 above)
Comment (garymm) [reply test/onnx/test_pytorch_onnx_no_runtime.py:13]: OperatorExportTypes is available in Python, you shouldn't need to access it via `torch._C`.

Comment (justinchuby) [reply test/onnx/test_pytorch_onnx_no_runtime.py:13]: True. I realized this is a test can thus should be considered to be outside of `torch.onnx`

File: test/onnx/test_pytorch_onnx_no_runtime.py [hunk 2]
```diff
@@ -107,8 +110,358 @@ def forward(self, y: Optional[Tensor]) -> Optional[Tensor]:
         )
 
 
-instantiate_parametrized_tests(TestOptionalOutput)
+# Below are moved from test/jit
```
Comment (garymm) [inline test/onnx/test_pytorch_onnx_no_runtime.py:113]: Why is this useful information for a future reader of this file?

File: test/test_jit.py [hunk 3]
```diff
@@ -1,91 +1,132 @@
 # -*- coding: utf-8 -*-
 # Owner(s): ["oncall: jit"]
 
-import torch
+import copy
```
Comment (garymm) [inline test/test_jit.py:4]: Remove any formatting changes so it's clear what the semantic changes are.
(Code outside of the onnx directories is not consistently formatted, and in general you should not auto-format it when modifying it).

**Review by garymm — CHANGES_REQUESTED** [review]
Congrats on your first PR!

**Comment by AllenTiTaiWang** [comment]
> @AllenTiTaiWang please sign the CLA as instructed by the bot

Done. Should be effective in an hour based on the feedback.

File: test/onnx/test_pytorch_onnx_no_runtime.py (hunk 1 above)
Comment (justinchuby) [inline test/onnx/test_pytorch_onnx_no_runtime.py:13]: use torch.onnx.OperatorExportTypes directly

Comment (justinchuby) [reply test/onnx/test_pytorch_onnx_no_runtime.py:13]: We should only import the test helper functions for now and build a good habit of keeping the namespace clean

**Review by justinchuby — APPROVED** [review]
LGTM

File: test/onnx/test_pytorch_onnx_no_runtime.py [hunk 4]
```diff
@@ -4,20 +4,22 @@
...
+from typing import Dict, Optional, Type
 
 import onnx
 
 import torch
 from torch import Tensor
-from torch.onnx import symbolic_helper
+from torch.onnx import OperatorExportTypes, symbolic_helper, utils
 from torch.testing._internal.common_utils import (
+    TestCase,
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply test/onnx/test_pytorch_onnx_no_runtime.py]: Looks like it sets the seed at setup and does other things. It seems to be commonly used across the package

https://github.com/pytorch/pytorch/blob/04ac80c73a9f525322a8b622659a27ad065698ea/torch/testing/_internal/common_utils.py#L1186-L1190

[the remark this answers is not in the document]
Comment (garymm) [reply test/onnx/test_pytorch_onnx_no_runtime.py]: If we're not using it in other tests in test/onnx, please don't introduce it here. We can separately decide if we want to adopt it for the entire directory.

**Comment by justinchuby** [comment]
> I think we should common_utils.TestCase's assertEqual in this case. It has the benefit of transparent error messages when the assertion fails, and it handles torch specific data types.

If `assert torch.all(torch.eq(result, reference))` fails, we will just see "False is not True`, which is not going to be very helpful.

**Comment by garymm** [comment]
> I think we should common_utils.TestCase's assertEqual in this case. It has the benefit of transparent error messages when the assertion fails, and it handles torch specific data types.

Agreed, that's a good reason to use the `common_utils.TestCase` :-)
Just didn't want to use it if we got no benefit.

**Review by garymm — APPROVED** [review]
👯

**Comment by justinchuby** [comment]
@AllenTiTaiWang you can ask the pytorchbot to merge it. I’ll leave your first merge to you

**Comment by garymm** [comment]
@malfet could you please import?

**Review by justinchuby — APPROVED** [review]

**Review by malfet — APPROVED** [review]
